kore/src/Kore/Step/Substitution.hs

```diff
@@ -128,58 +125,54 @@ normalizeSubstitutionAfterMerge
     -> BuiltinAndAxiomSimplifierMap level
     -> PredicateSubstitution level variable
     -> ExceptT
-          ( UnificationOrSubstitutionError level variable )
-          Simplifier
-          ( PredicateSubstitution level variable
-          , UnificationProof level variable
-          )
-normalizeSubstitutionAfterMerge
+          (UnificationOrSubstitutionError level variable)
+          (BranchT Simplifier)
+          (PredicateSubstitution level variable)
+normalizeWorker
     tools
     wrappedSimplifier@(PredicateSubstitutionSimplifier substitutionSimplifier)
     simplifier
     axiomIdToSimplifier
     Predicated { predicate, substitution }
   = do
-    (Predicated
-            { predicate = duplicationPredicate
-            , substitution = duplicationSubstitution
-            }
-        , proof
-        ) <-
-            normalizeSubstitutionDuplication' substitution
-
-    Predicated
-        { predicate = normalizePredicate
-        , substitution = normalizedSubstitution
-        } <- normalizeSubstitution' duplicationSubstitution
+    duplication <- normalizeSubstitutionDuplication' substitution
+    let
+        Predicated { substitution = duplicationSubstitution } = duplication
+        Predicated { predicate = duplicationPredicate } = duplication
 
+    normalized <- normalizeSubstitution' duplicationSubstitution
     let
+        Predicated { substitution = normalizedSubstitution } = normalized
+        Predicated { predicate = normalizedPredicate } = normalized
+
         mergedPredicate =
             makeMultipleAndPredicate
-                [predicate, duplicationPredicate, normalizePredicate]
-
-    (resultPredicateSubstitution, _proof) <-
-        lift $ substitutionSimplifier
-            Predicated
-                { term = ()
-                , predicate = mergedPredicate
-                , substitution = normalizedSubstitution
-                }
+                [predicate, duplicationPredicate, normalizedPredicate]
 
-    return
-        ( resultPredicateSubstitution
-        , proof
-        )
+    lift $ substitutionSimplifier
+        Predicated
+            { term = ()
+            , predicate = mergedPredicate
+            , substitution = normalizedSubstitution
+            }
   where
-    normalizeSubstitutionDuplication' =
-        normalizeSubstitutionDuplication
-            tools
-            wrappedSimplifier
-            simplifier
-            axiomIdToSimplifier
-    normalizeSubstitution' =
-        withExceptT substitutionToUnifyOrSubError
-            . normalizeSubstitution tools
+    normalizeSubstitutionDuplication' substitution' = do
+        (duplication, _) <-
+            Monad.Morph.hoist lift
```

Comment:
Does any of the 4 lift calls below need to use liftPruned?

Reply:
No. The outer calls to `lift` are used to put `BranchT Simplifier` under `ExceptT`. The `Monad.Morph.hoist lift` calls cannot use `liftPruned` instead due to the constraints imposed by `hoist`, so I `returnPruned` below instead.